src/classes/DRS_ContactMerge_CTRL.cls

```diff
@@ -0,0 +1,1036 @@
+/*
+    Copyright (c) 2020, Salesforce.org
+    All rights reserved.
+    
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020 (2.x)
+* @description Controller for the NPSP DRS contact merge page.
+* @group ContactMerge
+* @group-content ../../ApexDocContent/ContactMerge.htm
+*/
+
+public with sharing class DRS_ContactMerge_CTRL {
```

Comment:
I'm not sold on the "DRS" prefix for this class ... I think it belongs with the "CON" prefix but maybe "CON_DRSContactMerge_CTRL". That said, I'm seeing an awful lot of duplicated code between this class and CON_ContactMerge_CTRL. It really feels like these two should share a common base class that includes the behaviors they have in common, though I haven't had the chance to dive in deeply enough to make sure that's the best way to design this. As far as I can tell there are entire methods that are copy-pasted from one class to another however, and that needs to change.

Reply:
@ptewson-sfdo Contact merge logic implemented on the DRS Contact Merge page is same as that existing Contact Merge functionality.
We have replicated the loadMergeCandidates() method because it is declared private in CON_ContactMerge_CTRL class. Along with these we had to replicate the selectDefaultRecord(), mergeContacts() methods and its related wrapper classes as they are accessing some of the private members of CON_ContactMerge_CTRL class. We were not able to call these methods from DRS_ContactMerge_CTRL class as either they were declared private or were accessing any private members. We replicated the methods instead of creating a new base class because we wanted to make minimal changes to the already existing Contact Merge functionality code. Creating a new base class will cause a significant amount of changes to the already existing CON_ContactMerge_CTRL class. Please let us know if you want us to proceed with the new base class